### ISSUE TITLE
linter warns on item property of vaadin-grid templates

### ISSUE THREAD

**Issue by bennypowers** [Polymer/polymer-linter#83]
`<vaadin-grid>` uses the keyword `item` to represent iterated data points in it's grid templates, like so

```html
  <vaadin-grid aria-label="Two-way Binding Example" items="[[users.result]]">

    <vaadin-grid-column>
      <template class="header">First Name</template>
      <template>
        <input value="{{item.firstName::input}}" readonly$="[[!editing]]">
      </template>
    </vaadin-grid-column>

    <vaadin-grid-column>
      <template class="header">Last Name</template>
      <template>
        <input value="{{item.lastName::input}}" readonly$="[[!editing]]">
      </template>
    </vaadin-grid-column>

    <vaadin-grid-column width="200px">
      <template class="header">Email</template>
      <template>
        <div>[[item.firstName]].[[item.lastName]]@example.com</div>
      </template>
    </vaadin-grid-column>

  </vaadin-grid>
```

The above example does not trigger linting errors, however, if `[[item.prop]]` is bound in a child template, polymer linter warns.

```html
<vaadin-grid id="grid" name="grid" items="[[videos]]">

<vaadin-grid-column flex-grow="2">
  <template class="header">Title</template>
  <template>
    <span class="title">[[item.videoData.title]]</span>
    <template is="dom-if" if="[[!same(item.videoData.title, item.videoData.filename)]]" restamp="true">
      <span class="filename">[[item.videoData.filename]]</span>
    </template>
  </template>
</vaadin-grid-column>
```
in which case, the binding expression `[[item.videoData.filename]]` triggers a warning. ` `item is not declared or used more than once.`

**Comment by stramel** [comment]
I'm also getting 

```
      aria-label="[[gridDescription]]"
      ~~~~~~~~~~

mtz-group-list.html(33,6) warning [set-unknown-attribute] - vaadin-grid elements do not have a property named ariaLabel. Consider instead:  aria-label$
```

**Comment by web-padawan** [comment]
Please see #67

**Comment by stramel** [comment]
Hmmm #67 looks similar to the `items` issue however `ariaLabel` piece seems unrelated?

**Comment by web-padawan** [comment]
`aria-label` is not a property and therefore you should use attribute binfing there

**Comment by stramel** [comment]
You're right, maybe a better error message would be more useful.

**Comment by bennypowers** [comment]
I have the same error with this syntax:

```html
<template is="dom-repeat" items="[[array]]" as="clipset" index-as="index">
  <paper-listbox name$="[[_getClipsetName(index, clipsets)]]" selected="{{clipIndex}}">
    <template is="dom-repeat" items="[[clipset]]" as="clip" index-as="makesError">
      <paper-item id="[[clip.id]]" on-tap="_handleTap">
        <span class="index">[[makesError]]</span>
```

**Comment by bennypowers** [comment]
For anyone arriving from Google, welcome 👋 

In the meantime, I've disabled these warnings in CI by defining my lint script in my PSK app's package.json as such:
```js
"lint": {
  "rules": ["polymer-2-hybrid"],
  "ignoreWarnings": ["databind-with-unknown-property"]
},
```

**Comment by jarrodek** [comment]
I'm not sure if this is the same issue but I'm having similar error message when using template inside another template.
Iron list uses `<template>` to generate list item and inside this template I am using `item` property that is used inside another template:

```
<iron-list>
  <template>
    <div>
      <span>[[item.property]] - this works here fine</span>
      <template is="dom-if" if="[[item.property]]">
         <span>[[item.property]] - this works here too but linter shows error</span>
      </template>
    </div>
  </template>
<iron-list>
```

![image](https://user-images.githubusercontent.com/3320020/32409969-e46b1540-c173-11e7-9cfa-b9616615d2a5.png)
